src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -159,9 +159,62 @@ class spell_monk_provoke : public SpellScript
     }
 };
 
+// Roll - 109132
+class spell_monk_roll : public SpellScriptLoader
+{
+public:
+    spell_monk_roll() : SpellScriptLoader("spell_monk_roll") { }
+
+    class spell_monk_roll_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_monk_roll_SpellScript);
+
+        bool Validate(SpellInfo const* /*spell*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(109132, DIFFICULTY_NONE))
+                return false;
+            return true;
+        }
+
+        void HandleBeforeCast()
+        {
+            Aura* aur = GetCaster()->AddAura(107427, GetCaster());
+            if (!aur)
+                return;
+
+            AuraApplication* app = aur->GetApplicationOfTarget(GetCaster()->GetGUID());
+            if (!app)
+                return;
+
+            app->ClientUpdate();
+        }
+
+        void HandleAfterCast()
+        {
+            Unit* caster = GetCaster();
+            if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
+                return;
+
+            caster->CastSpell(caster, 107427, true);
+        }
+
+        void Register() override
+        {
+            BeforeCast += SpellCastFn(spell_monk_roll_SpellScript::HandleBeforeCast);
```

Comment:
1. Prefer to add events to effects rather than the entire spell. Most spells will have effects of type dummy, which can be a hint to use it.
2. In this case, using `OnEffectHit` would be a better choice; just cast a spell when the effect hit. No reason for the Before or After.